vncdotool/command.py

```diff
@@ -16,15 +15,20 @@
 import socket
 import logging
 import logging.handlers
+from types import TracebackType
+from typing import List, Optional, Tuple, Type, TypeVar
 
 from twisted.python.log import PythonLoggingObserver
 from twisted.internet import reactor, protocol
 from twisted.internet.error import ConnectionDone
+from twisted.internet.interfaces import IConnector
 from twisted.python.failure import Failure
 
 from .client import VNCDoToolFactory, VNCDoToolClient, factory_connect
 from .loggingproxy import VNCLoggingServerFactory
 
+V = TypeVar("V", bound="VNCDoCLIClient")
```

Comment:
Is using V an common idiom?  What is this actually doing?

Reply:
`V`, `K`, `VT`, `VK` are often used for _unbound_ `TypeVar`s, but as now it is _bound_ `V` is no longer a good name. Thank you for spotting this. I changed it to `vncdotool.client.TClient`.

[TypeVar](https://mypy.readthedocs.io/en/stable/generics.html#generics)s are used to bind some type to a name: Take this simple function for an example: `def same(var: V) -> V: return var`
You can pass any type (int, str, object, ….) into that function and will get the same type back.
The additional `bound="VNCDoCLIClient"` adds one more restriction: the passed in value must be an instance of `VNCDoCLIClient` (or any subclass of it) and you will (still) get the same type back. This is important if you actually pass in a sub-class of `VNCDoCLIClient` as then you're guaranteed to get back that sub-class, and not only the super-class as you would get with `def f(var: VNCDoCLIClient) -> VNCDoCLIClient: ...`: There you could still pass in a sub-class of `VNCDoCLIClient`, but you're only guaranteed to get back a `VNCDoCLIClient`, which might then not have the added methods of your sub-class.